src/PackageCollector.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { red } from 'chalk';
+import { join } from 'path';
+import { readdirSync, statSync } from 'fs';
+import logger from './logger';
+import PackageParser from './PackageParser';
+import {
+  EXTENSIONS_FOLDER,
+  EXTENSIONS_FOLDER_FRONTEND,
+} from './constants';
+
+/**
+ * The PackageCollector class.
+ */
+class PackageCollector {
+  /**
+   * Searches for packages within a list of passed paths.
+   * @param {Array|string} [paths] The paths where packages are located.
+   * @return {Array} The available packages.
+   */
+  get(paths = []) {
+    const sanitizedPaths = typeof paths === 'string' ? [paths] : paths;
+
+    let packagePaths = [];
+
+    // Collect all directories which might contain packages.
+    sanitizedPaths.forEach((path) => {
+      let subDirs;
+      if (path.split('/').includes(EXTENSIONS_FOLDER)) {
+        /**
+         * Special treatment for the the "extensions" folder,
+         * since it's structure differs from other folders.
+         */
+        subDirs = this.findSubDirectories(path, EXTENSIONS_FOLDER_FRONTEND);
+      } else {
+        subDirs = this.getSubDirectories(path);
+      }
+
+      packagePaths = packagePaths.concat(subDirs);
+    });
+
+    const packages = [];
```

Comment:
If you would use `.map()` as described below, you could get rid of this variable and directly return the return value of `.map()`.